services/identity-service/src/did/did.controller.ts

```diff
@@ -0,0 +1,69 @@
+import {
+  Body,
+  Controller,
+  Get,
+  NotFoundException,
+  Param,
+  Patch,
+  Post,
+  UseGuards,
+} from '@nestjs/common';
+import {
+  ApiBadRequestResponse,
+  ApiBody,
+  ApiCreatedResponse,
+  ApiNotFoundResponse,
+  ApiOkResponse,
+  ApiOperation,
+  ApiParam,
+  ApiResponse,
+  ApiTags,
+} from '@nestjs/swagger';
+import { DIDDocument } from 'did-resolver';
+import { DidService } from './did.service';
+import { GenerateDidDTO } from './dtos/GenerateDid.dto';
+import { JwtAuthGuard } from './roles.guard';
+
+@ApiTags('DID')
+@Controller('did')
+export class DidController {
+  constructor(private readonly didService: DidService) {}
+
+  @ApiOperation({ summary: 'Generate a new DID' })
```

Comment:
Exception handling is missing? How are we handling errors, validations, exceptions.